doc/rtd/reference/datasources/none.rst

```diff
@@ -0,0 +1,51 @@
+.. _datasource_none:
+
+None
+****
+
+The data source ``None`` may be used when no other viable datasource is
+present on disk. This has three primary use cases:
+
+1. Testing cloud-init in an environment where no datasource is
+   available.
+2. As a fallback for when a datasource is otherwise intermittently
+   unavailable.
+3. Providing user data to cloud-init from on-disk configuration when
+   no other datasource is present.
+
+When the datasource is ``None``, cloud-init is unable to obtain or
+render networking configuration. Additionally, when cloud-init
+completes, a warning is logged that DataSourceNone is being used.
```

Comment:
> a warning is logged that DataSourceNone is being used.

Is it logged if DatasourceNone is being used, or is it logged if this datasource is used as a fallback? I don't think that this is behavior that we actually want if only [None] is specified by the user, right?

Reply:
It happens regardless.

https://github.com/canonical/cloud-init/blob/main/cloudinit/config/cc_final_message.py#L114

I can create an issue to remove it if None has userdata/metadata or is the only entry in the datasource list.

Reply:
sounds good to me!

Reply:
https://github.com/canonical/cloud-init/issues/5192